StardewArchipelago/Locations/CodeInjections/Modded/MagicModInjections.cs

```diff
@@ -0,0 +1,228 @@
+    using System;
+using System.Reflection;
+using System.Collections.Generic;
+using Netcode;
+using HarmonyLib;
+using Microsoft.Xna.Framework;
+using StardewModdingAPI;
+using StardewArchipelago.Archipelago;
+using StardewArchipelago.Items.Unlocks;
+using StardewValley;
+using StardewValley.Menus;
+
+namespace StardewArchipelago.Locations.CodeInjections.Modded
+{
+    public static class MagicModInjections
+    {
+        private const string LEARN_BLINK_AP_LOCATION = "Analyze All Toil School Locations";
+        private const string LEARN_SPIRIT_AP_LOCATION = "Analyze All Eldritch School Locations";
+        private const string LEARN_CLEARDEBRIS_AP_LOCATION = "Analyze: Clear Debris";
+        private const string LEARN_TILL_AP_LOCATION = "Analyze: Till";
+        private const string LEARN_WATER_AP_LOCATION = "Analyze: Water";
+        private const string LEARN_EVAC_AP_LOCATION = "Analyze: Evac";
+        private const string LEARN_HASTE_AP_LOCATION = "Analyze: Haste";
+        private const string LEARN_HEAL_AP_LOCATION = "Analyze: Heal";
+        private const string LEARN_BUFF_AP_LOCATION = "Analyze All Life School Locations";
+        private const string LEARN_SHOCKWAVE_AP_LOCATION = "Analyze: Shockwave";
+        private const string LEARN_FIREBALL_AP_LOCATION = "Analyze: Fireball";
+        private const string LEARN_FROSTBITE_AP_LOCATION = "Analyze: Frostbite";
+        private const string LEARN_TELEPORT_AP_LOCATION = "Analyze All Elemental School Locations";
+        private const string LEARN_LANTERN_AP_LOCATION = "Analyze: Lantern";
+        private const string LEARN_TENDRILS_AP_LOCATION = "Analyze: Tendrils";
+        private const string LEARN_PHOTOSYNTHESIS_AP_LOCATION = "Analyze All Nature School Locations";
+        private const string LEARN_DESCEND_AP_LOCATION = "Analyze: Descend";
+        private const string LEARN_METEOR_AP_LOCATION = "Analyze: Meteor";
+        private const string LEARN_LUCKSTEAL_AP_LOCATION = "Analyze: Lucksteal";
+        private const string LEARN_BLOODMANA_AP_LOCATION = "Analyze: Bloodmana";
+        private const string LEARN_REWIND_AP_LOCATION = "Analyze Every Magic School Location";
+
+        private static IMonitor _monitor;
+        private static IModHelper _helper;
+        private static ArchipelagoClient _archipelago;
+        private static LocationChecker _locationChecker;
+
+
+        public static void Initialize(IMonitor monitor, IModHelper modHelper, ArchipelagoClient archipelago, LocationChecker locationChecker)
+        {
+            _monitor = monitor;
+            _helper = modHelper;
+            _archipelago = archipelago;
+            _locationChecker = locationChecker;
+        }
+
+        public static bool OnCast_AnalyzeGivesLocations_Prefix(Farmer player, int level, int targetX, int targetY)
```

Comment:
Okay so this method is **gigantic**. Please split it up in a bunch of smaller ones. I can see, at a glance, at least 5 smaller ones, probably more